### PR TITLE
738 update derive var naming convention

### PR DESCRIPTION
**Pull request by amsmith214**
Functions, examples, example scripts, vignettes are updated for 

- `derive_disposition_dt()`, `derive_disposition_status()`, `derive_extreme_flag`, `derive_worst_flag`, `derive_last_dose`, `derive_obs_number`, `derive_disposition_reason` 

to reflect naming changes to 

- `derive_var_disposition_dt()`, `derive_var_disposition_status()`, `derive_var_extreme_flag`, `derive_var_worst_flag`, `derive_var_last_dose`, `derive_var_obs_number`, `derive_vars_disposition_reason` 

respectively. Deprecation warnings added to documentation for affected functions. Sorry for the deletion/addition, this seems like an artifact of my machine's Rstudio git interface being incredibly slow.

### CODE REVIEW

**Comment by amsmith214** [comment]
will fix the failed check tomorrow

**Review by bundfussr — CHANGES_REQUESTED** [review]
Nice work!

- `derive_agegr_fda()` and  `derive_agegr_ema()` should be renamed as well.
- Line 228 in adsl.Rmd should be updated. ("derive_disposition_status(")
- Line 297 in adsl.Rmd should be updated. ("derive_disposition_reason(")
- Line 540 in bds_finding.Rmd should be updated. ("Note: Additional examples of the `derive_extreme_flag()` function can be")
- Line 581in bds_tte.Rmd should be updated. ("The admiral function `derive_obs_number()` can be used to derive `ASEQ`:")
- Line 234 in occds.Rmd should be updated. ("The function `derive_last_dose()` can be used to derive the last dose date before the start of the event. ")
- Line 349 in occds.Rmd should be updated. ("The function `derive_extreme_flag()` can help derive variables such as `AOCCIFL`,")
- Line 101in unit_test_guidance.Rmd should be updated. ("your function justifies it, e.g. see the test script: https://github.com/Roche-GSK/admiral/blob/master/tests/testthat/test-derive_extreme_flag.R.")

**Comment by amsmith214** [comment]
> Nice work!
> 
> * `derive_agegr_fda()` and  `derive_agegr_ema()` should be renamed as well.
> * Line 228 in adsl.Rmd should be updated. ("derive_disposition_status(")
> * Line 297 in adsl.Rmd should be updated. ("derive_disposition_reason(")
> * Line 540 in bds_finding.Rmd should be updated. ("Note: Additional examples of the `derive_extreme_flag()` function can be")
> * Line 581in bds_tte.Rmd should be updated. ("The admiral function `derive_obs_number()` can be used to derive `ASEQ`:")
> * Line 234 in occds.Rmd should be updated. ("The function `derive_last_dose()` can be used to derive the last dose date before the start of the event. ")
> * Line 349 in occds.Rmd should be updated. ("The function `derive_extreme_flag()` can help derive variables such as `AOCCIFL`,")
> * Line 101in unit_test_guidance.Rmd should be updated. ("your function justifies it, e.g. see the test script: https://github.com/Roche-GSK/admiral/blob/master/tests/testthat/test-derive_extreme_flag.R.")

Thanks, and sorry for missing so many instances. Do we have an elegant way of finding instances of deprecated function names in the documentation, or is it down to using the GH search?

**Comment by bundfussr** [comment]
I used `grep -P 'derive_(obs|last|wor|disp|extr)' *.Rmd` in the terminal.

**Comment by amsmith214** [comment]
> I used `grep -P 'derive_(obs|last|wor|disp|extr)' *.Rmd` in the terminal.

regex to the rescue!

File: R/derive_var_disposition_dt.R
Comment (amsmith214) [reply R/derive_var_disposition_dt.R]: done and tested on examples from documentation

File: R/derive_var_disposition_status.R
Comment (amsmith214) [reply R/derive_var_disposition_status.R:115]: done and tested on examples from documentation

File: R/derive_var_extreme_flag.R
Comment (amsmith214) [reply R/derive_var_extreme_flag.R:166]: done and tested on examples from documentation

File: R/derive_var_extreme_flag.R
Comment (amsmith214) [reply R/derive_var_extreme_flag.R]: warning added

File: R/derive_var_last_dose.R
Comment (amsmith214) [reply R/derive_var_last_dose.R:111]: done and tested on examples from documentation

File: R/derive_var_obs_number.R
Comment (amsmith214) [reply R/derive_var_obs_number.R]: done and tested on examples from documentation

File: R/derive_vars_disposition_reason.R
Comment (amsmith214) [reply R/derive_vars_disposition_reason.R:148]: done and tested on examples from documentation

**Comment by thomas-neitmann** [comment]
There are several lintr annotations. Please fix these issues (you can run `lintr::lint_package()` locally to get a nice overview). Otherwise this looks good to me 👍

**Comment by thomas-neitmann** [comment]
@amsmith214 There's a merge conflict that needs to be taken care off before we can merge this PR into `devel`. Please resolve locally in RStudio. The git guidance vignette should be helpful. Please refer to the version on the `devel` branch.

**Review by thomas-neitmann — APPROVED** [review]

**Comment by thomas-neitmann** [comment]
@bundfussr Please approve as well if this looks good to you. Without your approval merging is currently blocked.

**Review by bundfussr — APPROVED** [review]

**Pull request merged**
